python/tvm/relay/frontend/pytorch.py

```diff
@@ -0,0 +1,1138 @@
+                self._nid_to_node_name[op_name] = nid
+                nid = nid+1
+
+        outputs = []
+
+        for i in range(nid):
+            output = self._relay_map[i]
+            outputs.append(output)
+
+        if len(outputs) == 1:
+            body = outputs[0]
+        else:
+            body = outputs[-1]
+
+        func = tvm.relay.Function(_analysis.free_vars(body), body)
+
+        param = {k: tvm.nd.array(v) for k, v in self._param_tensors.items()}
+
+        return  _module.Module.from_expr(func), param
+
+    def _parse_inputs(self):
+        """ Map inputs to parser and inputs to graph. """
+        # Get names and objects of inputs for IR
+        ir_names = [i.debugName() for i in self._trace.graph.inputs()]
+        ir_inputs = [i for i in self._trace.graph.inputs()]
+
+        # Create corresponding shape and add to input
+        for input_name, ir_input in zip(self._input_shapes, ir_inputs[1:]):
+            input_shape = self._input_shapes[input_name]
+            ir_input.setDebugName(input_name)
+
+            self._inputs_r[input_name] = _expr.var(input_name,
+                                                   shape=self._input_shapes[input_name],
+                                                   dtype=_convert_data_type(
+                                                       self._input_types[input_name]))
+            self._fn_param.append(_expr.var(input_name,
+                                            shape=self._input_shapes[input_name],
+                                            dtype=_convert_data_type(
+                                                self._input_types[input_name])))
+
+        # Add self (first input of a PyTorch graph) to inputs
+        input_shape = [3]
+        tensor = tvm.nd.array(np.zeros(input_shape).astype(np.float32))
+        input_name = ir_names[0]
+        self._inputs_r[input_name] = tensor
+
+    def _parse_params(self):
+        """ Map state dictionary values to corresponding prim::GetAttr op node. """
+        # Grab weights, biases, etc. from graph
+        state_dict = self._trace.state_dict()
+        param_names = []
+        for key, value in state_dict.items():
+            param_str = str(key)
+            param_name = param_str.split('.')[-1]
+            param_names.append(param_name)
+
+        # Get names of all inputs
+        input_names = [i for i in self._inputs_r.keys()]
+
+        # Iterate through graph for getAttr nodes and match full state_dict name to nodes
+        node_weight_map = {}
+        for node in self._trace.graph.nodes():
+            if node.kind() == "prim::GetAttr":
+                node_str = str(node)
+                node_assign = (node_str.split(' = ')[0]).split(' : ')
+                node_name = (node_assign[0])[1:]
+                node_getattr_name = ((node_str.split(' = ')[1]).split('"')[1::2])[0]
+                node_arg = (((node_str.split(' = '))[1]).split('(')[1])[1:-2]
+
```

Comment:
we should be able to use PyTorch's API to retrieve these information, rather than string processing.
For example, 
* node_name -> node.output() (torch._C.Value)
* list of attribute names -> node.attributeNames() (on GetAttr, it will return ["name"])
* node_getattr_name -> node.s("name") (refer to https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/ir.cpp#L152-L153)
* node_getattr_name -> node.
* node_arg -> node.input() (torch._C.Value)

Reply:
you can replace string stuff with something like below
```
if node.kind() == "prim::GetAttr":
    attribute_names = node.attributeNames()
    assert(len(attribute_names) == 1)
    node_getattr_name = node.s(attribute_names[0])
    node_arg = node.input().debugName()
    node_name = node.output().debugName()
```